go/database/mpt/forest_test.go

```diff
@@ -561,7 +628,19 @@ func TestForest_getSharedNode_Fails_Get_Copy(t *testing.T) {
 					if err != nil {
 						t.Fatalf("failed to open forest: %v", err)
 					}
-					defer forest.Close()
+					// backup stocks and re-assign them for proper close
+					accountsBackup := forest.accounts
+					valuesBackup := forest.values
+
+					defer func() {
+						forest.accounts = accountsBackup
+						forest.values = valuesBackup
+					}()
+					defer func() {
+						if err = forest.Close(); err != nil {
+							t.Fatalf("failed to close forest: %v", err)
+						}
+					}()
```

Comment:
In this setup the `forest` is closed before the backups are restored. Is this intended? If so, please comment why the backup is still needed in this case.

Reply:
Thanks for noticing